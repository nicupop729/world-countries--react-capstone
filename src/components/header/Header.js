import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import _ from 'lodash';
import './header.css';

const Header = ({ onSearchCountry }) => {
  const [search, setSearch] = useState('');

  const continents = useSelector((state) => state);

  const countries = Object.values(continents);
  const finalC = [];
  countries.map((continent) => finalC.push(...continent));

  useEffect(() => {
    const s = search.toLowerCase();
    const searchCountry = finalC.filter((country) => country.name.common.toLowerCase().includes(s));
    const sortedFiteredCountries = _.sortBy(searchCountry, ['name.common'], ['asc']);
    onSearchCountry(sortedFiteredCountries);
  }, [search]);

  const countrySearchHandler = (e) => {
    setSearch(e.target.value);
  };

  return (
    <header className="Header">
      <h1 className="Header__title">World Countries</h1>
      <form>
        <input
          id="search-bar"
          type="text"
          placeholder="Search country"
          onChange={countrySearchHandler}
          value={search}
          className="input-searh"
        />
      </form>
    </header>
  );
};

Header.propTypes = {
  onSearchCountry: PropTypes.func.isRequired,
};

export default Header;
