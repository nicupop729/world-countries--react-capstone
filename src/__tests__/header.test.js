import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Header from '../components/header/Header';

describe('matches snapshot', () => {
  test('matches to snapshot', () => {
    const handlerSearchCountry = jest.fn();
    const tree = renderer.create(
      <Router>
        <Provider store={store}>
          <Header onSearchCountry={handlerSearchCountry} />
        </Provider>
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
