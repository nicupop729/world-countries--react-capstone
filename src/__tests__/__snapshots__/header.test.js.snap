// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot matches to snapshot 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__title"
  >
    World Countries
  </h1>
  <form>
    <input
      className="input-searh"
      id="search-bar"
      onChange={[Function]}
      placeholder="Search country"
      type="text"
      value=""
    />
  </form>
</header>
`;
